src/Avalonia.Visuals/Rendering/SceneGraph/Scene.cs

```diff
@@ -150,7 +150,7 @@ private VisualNode Clone(VisualNode source, IVisualNode parent, Dictionary<IVisu
 
             index.Add(result.Visual, result);
 
-            foreach (var child in source.Children)
+            foreach (var child in source.Children.Where(x=>!x.Disposed))
```

Comment:
Best not to use in rendering code for performance reasons. Put an `if` inside the `foreach` instead.

Reply:
Why is it possible that a disposed object is still in the Children collection? Isn't that the real cause? Hiding the original cause is probably a bad idea.